src/util.py

```diff
@@ -4,27 +4,18 @@
 def printf(*args):
     print(*args, flush=True)
 
+def bitFill(count):
+    return (1 << count) - 1
 
-def bitGet(bits, bitStart, bitCount):
-    value = 0
-    for i in range(bitCount):
-        bit = 1 << (bitStart + i)
-        if bits & bit:
-            value |= 1 << i
-
-    return value
-
-
-def bitSet(bits, value, bitStart, bitCount):
-    for i in range(bitCount):
-        bit = 1 << i
-        if value & bit:
-            bits |= 1 << (bitStart + i)
-        else:
-            bits &= ~(1 << (bitStart + i))
-
+def bitGet(bits, startBit, count):
+    bits = bits >> startBit
+    bits &= bitFill(count)
     return bits
 
+def bitSet(bits, sourceBits, startBit, count):
+    sourceBits = bitGet(sourceBits, 0, count) << startBit
```

Comment:
Is 0 correct here? It seems like that wouldn't get you the right set of bits

Reply:
Rather, the value passed in shouldn't be shifted at all, it should just be used as-is. So that settings bits 5-8 to 15 only requires passing `(bits, 15, 4, 4)` rather than shifting value beforehand.